From 29ae79e777b1b8d8115127d47c902c76d8ce0c6e Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Fri, 09 Aug 2024 11:13:54 +0300
Subject: [PATCH] [SW-0]: fs1: run reset arc from iccm on pldm

This patch is to build reset zephyr-apps
for only iccm. This is needed to load
reset fw as standalone application on PLDM

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ibc5d2bcb2af8160a8057e88fbeab6a55ce3d566e
---

diff --git a/boards/intel/fs/fs1_rst_onix_defconfig b/boards/intel/fs/fs1_rst_onix_defconfig
index acec5ae..346692d 100644
--- a/boards/intel/fs/fs1_rst_onix_defconfig
+++ b/boards/intel/fs/fs1_rst_onix_defconfig
@@ -17,3 +17,4 @@
 CONFIG_MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE=n
 CONFIG_BOOT_IF_PSEUDO_RO_TIMER=y
 CONFIG_XIP=n
+CONFIG_RST_RUN_FROM_ICCM=y
diff --git a/soc/intel/gpu/fs1/emb/Kconfig.defconfig_rst b/soc/intel/gpu/fs1/emb/Kconfig.defconfig_rst
index 7bffdb4..8473e0e 100644
--- a/soc/intel/gpu/fs1/emb/Kconfig.defconfig_rst
+++ b/soc/intel/gpu/fs1/emb/Kconfig.defconfig_rst
@@ -7,7 +7,7 @@
 	default "fs1_rst"
 
 config HARVARD
-	default y
+	default n
 
 config NUM_IRQS
 	range 17 255
@@ -18,4 +18,10 @@
 config SHARED_IRQ
 	default y
 
+config RST_RUN_FROM_ICCM
+	default n
+	bool "link dsm app to run from ICCM memory"
+	help
+	  Special emulator mode where dsm app is executing from ICCM memory
+
 endif # SOC_FS1_RST
diff --git a/soc/intel/gpu/fs1/emb/fs1_linker.ld b/soc/intel/gpu/fs1/emb/fs1_linker.ld
index b170710..2aacc46 100644
--- a/soc/intel/gpu/fs1/emb/fs1_linker.ld
+++ b/soc/intel/gpu/fs1/emb/fs1_linker.ld
@@ -32,7 +32,7 @@
 
 #else /* CONFIG_HARVARD */
 
-	#if defined(CONFIG_PREBOOT_RUN_FROM_ICCM) || defined(CONFIG_MGMT_RUN_FROM_ICCM) || defined(CONFIG_DSM_RUN_FROM_ICCM)
+	#if defined(CONFIG_PREBOOT_RUN_FROM_ICCM) || defined(CONFIG_MGMT_RUN_FROM_ICCM) || defined(CONFIG_DSM_RUN_FROM_ICCM) || defined(CONFIG_RST_RUN_FROM_ICCM)
 		#if DT_NODE_HAS_PROP(DT_INST(0, arc_iccm), reg) && \
 			(DT_REG_SIZE(DT_INST(0, arc_iccm)) > 0)
 		#define SRAM_START		DT_REG_ADDR(DT_INST(0, arc_iccm))
